eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/tagwrite/BufferedTagWriteManager.java

```diff
@@ -0,0 +1,432 @@
+    } else {
+      writeTagBuffered(
+          managedTagProvider, connectorSettings, m2WebEwon, m2WebEwonEBDTag, tagPath, tagValue);
+    }
+  }
+
+  /**
+   * Processes an immediate tag write using the specified parameters. This method will write the tag
+   * immediately, without buffering.
+   *
+   * @param managedTagProvider the managed tag provider where the tag is located
+   * @param connectorSettings the connector settings
+   * @param m2WebEwon the M2Web Ewon object for the Ewon where the tag is located
+   * @param m2WebEwonEBDTag the M2Web Ewon EBD tag object for the tag to write
+   * @param tagPath the managed tag provider tag path
+   * @param tagValue the tag value to write
+   * @since 1.0.0
+   */
+  private static void writeTagImmediately(
+      ManagedTagProvider managedTagProvider,
+      EwonConnectorSettings connectorSettings,
+      M2WebEwon m2WebEwon,
+      M2WebEwonEBDTag m2WebEwonEBDTag,
+      TagPath tagPath,
+      Object tagValue) {
+    // Create tag write callback
+    FutureCallback<M2WebEwonUpdateTagValuesResponse> updateTagValuesResponseFutureCallback =
+        new FutureCallback<>() {
+          @Override
+          public void completed(M2WebEwonUpdateTagValuesResponse m2WebEwonUpdateTagValuesResponse) {
+            // Update tag value in managed tag provider
+            managedTagProvider.updateValue(tagPath.toString(), tagValue, QualityCode.Good);
+
+            // Log debug message
+            if (connectorSettings.isDebugEnabled()) {
+              LOGGER.debug(
+                  "Tag write performed (unbuffered) for tag [{}] with value [{}].",
+                  tagPath,
+                  tagValue.toString());
+            }
+          }
+
+          @Override
+          public void failed(Exception e) {
+            // Update tag value in managed tag provider
+            managedTagProvider.updateValue(
+                tagPath.toString(), tagValue, QualityCode.Error_Exception);
+
+            // Log error message
+            LOGGER.error(
+                "Tag write failed (unbuffered) for tag [{"
+                    + tagPath
+                    + "}] with value [{"
+                    + tagValue.toString()
+                    + "}].",
+                e);
+          }
+
+          @Override
+          public void cancelled() {
+            // Update tag value in managed tag provider
+            managedTagProvider.updateValue(
+                tagPath.toString(), tagValue, QualityCode.Error_ScriptEval);
+
+            // Log error message
+            LOGGER.error(
+                "Tag write cancelled (unbuffered) for tag [{}] with value [{}].",
+                tagPath,
+                tagValue.toString());
+          }
+        };
+
+    // Write tag immediately
+    List<KeyValue<String, Object>> tagNameValuePairs =
+        List.of(new DefaultKeyValue<>(m2WebEwonEBDTag.getName(), tagValue));
+    M2WebCommunicationManager.updateEwonTagValues(
```

Comment:
Just checking that this doesn't need a try/catch. I gather that failed, cancelled methods will be called.

Reply:
No need. There are no exceptions thrown in `updateEwonTagValues(...)`, either checked or unchecked.